### PR TITLE
removing docker-credentials secret

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: SHLOMITK@il.ibm.com

Fixes https://github.com/IBM/the-mesh-for-data/issues/210
- Removed the secret generation (docker-credentials)
- Removed attempts to login to the registry inside blueprint controller
- Removed references to docker variables inside motion webhooks

### CODE REVIEW

**Review by roee88 — APPROVED** [review]

**Comment by hunchback** [comment]
@shlomitk1 - please revert the parts using the DOCKER_XXX in the controllers, this is not only useful for private registry but also for integration testing

**Review by hunchback — CHANGES_REQUESTED** [review]
reverts code in the controllers

File: manager/apis/motion/v1alpha1/batchtransfer_webhook.go [hunk 1]
```diff
@@ -45,13 +46,6 @@ func (r *BatchTransfer) Default() {
 	if r.Spec.Image == "" {
 		if env, b := os.LookupEnv("MOVER_IMAGE"); b {
 			r.Spec.Image = env
-		} else {
```
Comment (hunchback) [inline manager/apis/motion/v1alpha1/batchtransfer_webhook.go:48]: revert

File: manager/apis/motion/v1alpha1/streamtransfer_webhook.go [hunk 2]
```diff
@@ -33,13 +34,6 @@ func (r *StreamTransfer) Default() {
 	if r.Spec.Image == "" {
 		if env, b := os.LookupEnv("MOVER_IMAGE"); b {
 			r.Spec.Image = env
-		} else {
```
Comment (hunchback) [inline manager/apis/motion/v1alpha1/streamtransfer_webhook.go:36]: revert

File: manager/controllers/app/blueprint_controller.go [hunk 3]
```diff
@@ -163,18 +161,6 @@ func (r *BlueprintReconciler) applyChartResource(log logr.Logger, chartSpec app.
 	nbytes, _ := yaml.Marshal(args)
 	log.Info(fmt.Sprintf("--- Values.yaml ---\n\n%s\n\n", nbytes))
 
-	// TODO: should change to use an ImagePullSecret referenced from the M4DModule resource
-	hostname := os.Getenv("DOCKER_HOSTNAME")
```
Comment (hunchback) [inline manager/controllers/app/blueprint_controller.go:167]: revert

**Comment by shlomitk1** [comment]
> 
> 
> @shlomitk1 - please revert the parts using the DOCKER_XXX in the controllers, this is not only useful for private registry but also for integration testing

How is this useful for integration testing? hostname is only used for login to the registry. Since we don't login to private registry during integration tests, why would it matter?

File: manager/controllers/app/blueprint_controller.go (hunk 3 above)
Comment (shlomitk1) [reply manager/controllers/app/blueprint_controller.go:167]: Why?

**Comment by hunchback** [comment]
> > @shlomitk1 - please revert the parts using the DOCKER_XXX in the controllers, this is not only useful for private registry but also for integration testing
> 
> How is this useful for integration testing? hostname is only used for login to the registry. Since we don't login to private registry during integration tests, why would it matter?
> @hunchback Please provide comments inside #210 to continue discussion on what should be done.

As contra, consider the following, hostname is not just used for login:

```
                        hostname, b1 := os.LookupEnv("DOCKER_HOSTNAME")	
			namespace, b2 := os.LookupEnv("DOCKER_NAMESPACE")	
			tagname, b3 := os.LookupEnv("DOCKER_TAGNAME")	
			if b1 && b2 && b3 {	
				r.Spec.Image = hostname + "/" + namespace + "/mover:" + tagname	
			}
```

File: manager/controllers/app/blueprint_controller.go (hunk 3 above)
Comment (hunchback) [reply manager/controllers/app/blueprint_controller.go:167]: ok, fine, this can be removed

**Comment by shlomitk1** [comment]
> 
> 
> > > @shlomitk1 - please revert the parts using the DOCKER_XXX in the controllers, this is not only useful for private registry but also for integration testing
> > 
> > 
> > How is this useful for integration testing? hostname is only used for login to the registry. Since we don't login to private registry during integration tests, why would it matter?
> > @hunchback Please provide comments inside #210 to continue discussion on what should be done.
> 
> As contra, consider the following, hostname is not just used for login:
> 
> ```
>                         hostname, b1 := os.LookupEnv("DOCKER_HOSTNAME")	
> 			namespace, b2 := os.LookupEnv("DOCKER_NAMESPACE")	
> 			tagname, b3 := os.LookupEnv("DOCKER_TAGNAME")	
> 			if b1 && b2 && b3 {	
> 				r.Spec.Image = hostname + "/" + namespace + "/mover:" + tagname	
> 			}
> ```

Unnecessary. Mover image can be constructed using MOVER_IMAGE env var, and according to @froesef we don't need yet another way.

**Review by hunchback — APPROVED** [review]

**Comment by froesef** [comment]
> > > > @shlomitk1 - please revert the parts using the DOCKER_XXX in the controllers, this is not only useful for private registry but also for integration testing
> > > 
> > > 
> > > How is this useful for integration testing? hostname is only used for login to the registry. Since we don't login to private registry during integration tests, why would it matter?
> > > @hunchback Please provide comments inside #210 to continue discussion on what should be done.
> > 
> > 
> > As contra, consider the following, hostname is not just used for login:
> > ```
> >                         hostname, b1 := os.LookupEnv("DOCKER_HOSTNAME")	
> > 			namespace, b2 := os.LookupEnv("DOCKER_NAMESPACE")	
> > 			tagname, b3 := os.LookupEnv("DOCKER_TAGNAME")	
> > 			if b1 && b2 && b3 {	
> > 				r.Spec.Image = hostname + "/" + namespace + "/mover:" + tagname	
> > 			}
> > ```
> 
> Unnecessary. Mover image can be constructed using MOVER_IMAGE env var, and according to @froesef we don't need yet another way.

I think the way to go here for customization is what @hunchback started at some point. To make it possible to define this in the module definition. (thus to customize the deployed helm charts in the module definition) (just fyi. This is independent of this issue)

**Review by froesef — APPROVED** [review]

**Pull request merged**